### PR TITLE
Octave: bump

### LINKED ISSUE #4538: Octave is broken
Octave is broken on latest Haiku revs (or at least on my hrev53658, Octave was worked before).

this is an output:
runtime_loader: /boot/system/lib/liboctave.so.4.0.0: Could not resolve symbol 'zgetri_'
resolve symbol "zgetri_" returned: -2147478780
runtime_loader: /boot/system/lib/liboctave.so.4.0.0: Troubles relocating: Symbol not found

### PR DESCRIPTION
**Pull request by extrowerk**
fixes #4538

### CODE REVIEW

File: sci-mathematics/octave/octave-5.1.0.recipe [hunk 1]
```diff
@@ -3,16 +3,21 @@ DESCRIPTION="GNU Octave is a high-level language, primarily intended for \
...
+CHECKSUM_SHA256="e36b1124cac27c7caa51cc57de408c31676d5f0096349b4d50b57bfe1bcd7495"
 PATCHES="octave-$portVersion.patchset"
 
 ARCHITECTURES="!x86_gcc2 x86 x86_64"
 SECONDARY_ARCHITECTURES="?x86"
 
+liboctaveVersion="7.0.0"
+liboctaveVersionCompat="$liboctaveVersion compat >= ${liboctaveVersion%%.*}"
+liboctinterpVersion="7.0.0"
+liboctinterpVersionCompat="$liboctinterpVersion compat >= ${liboctinterpVersion%%.*}"
```
Comment (Begasus) [inline sci-mathematics/octave/octave-5.1.0.recipe]: If both versions are the same you can just use libVersion and libVersionCompat?

File: sci-mathematics/octave/octave-5.1.0.recipe [hunk 2]
```diff
@@ -23,16 +28,23 @@ PROVIDES="
...
 	cmd:octave_config$secondaryArchSuffix = $portVersion
 	cmd:octave_config_$portVersion$secondaryArchSuffix = $portVersion
-	lib:liboctave$secondaryArchSuffix = $portVersion
-	lib:liboctinterp$secondaryArchSuffix = $portVersion
+	lib:liboctave$secondaryArchSuffix = $liboctaveVersionCompat
+	lib:liboctinterp$secondaryArchSuffix = $liboctinterpVersionCompat
 	"
 REQUIRES="
 	haiku$secondaryArchSuffix
 	cmd:gnuplot$secondaryArchSuffix
```
Comment (Begasus) [inline sci-mathematics/octave/octave-5.1.0.recipe]: There is no $secondaryArchSuffix for gnuplot
https://github.com/haikuports/haikuports/blob/b82dc69cb6d8cb3e8bb4f327473950587c8190bb/sci-visualization/gnuplot/gnuplot-5.2.5.recipe#L25

**Comment by korli** [comment]
maybe sundials could be enabled (https://computation.llnl.gov/casc/sundials/main.html) :)

**Comment by extrowerk** [comment]
@korli : sundials compiles fine on 64 bit, all test passes, will do a recipe for it, and extend the octave recipe accordingly.
As i can't give an exact timeframe for that because IRL things, i would say let us merge the recipe now and will do a revision with sundials later, as a user complained about broken octave.

File: sci-mathematics/octave/octave-5.1.0.recipe (hunk 2 above)
Comment (extrowerk) [reply sci-mathematics/octave/octave-5.1.0.recipe]: thx

File: sci-mathematics/octave/octave-5.1.0.recipe [hunk 3]
```diff
@@ -94,32 +118,45 @@ BUILD_REQUIRES="
...
 	devel:libsndfile$secondaryArchSuffix
+	devel:libsuitesparseconfig$secondaryArchSuffix
+	devel:libumfpack$secondaryArchSuffix
 	devel:libz$secondaryArchSuffix
 	"
 BUILD_PREREQUIRES="
+	cmd:autoconf
 	cmd:awk
 	cmd:bison >= 3
+#	cmd:qcollectiongenerator$secondaryArchSuffix >=5 # Qt
```
Comment (korli) [inline sci-mathematics/octave/octave-5.1.0.recipe]: This should be after cmd:python3

File: sci-mathematics/octave/octave-5.1.0.recipe [hunk 4]
```diff
@@ -94,32 +118,45 @@ BUILD_REQUIRES="
...
+	cmd:python3
+#	cmd:qhelpgenerator$secondaryArchSuffix >=5 # Qt
 #	cmd:rsvg_convert
 	cmd:sed
 	"
 
 BUILD()
 {
+	# Avoid using rsvg-convert and icotool, which we don't have
+	# and it doesn't make any sense to create thoose icons
```
Comment (korli) [inline sci-mathematics/octave/octave-5.1.0.recipe]:
```suggestion
	# and it doesn't make any sense to create those icons
```

**Comment by korli** [comment]
> i would say let us merge the recipe now and will do a revision with sundials later, as a user complained about broken octave.

Sure, I just don't know the order of merge between these PRs :)

**Comment by extrowerk** [comment]
Will do the merge then later.

File: sci-mathematics/octave/octave-5.1.0.recipe (hunk 1 above)
Comment (extrowerk) [reply sci-mathematics/octave/octave-5.1.0.recipe]: True, but the current solution is also ok.

**Pull request merged**